libnavui-tripprogress/src/main/java/com/mapbox/navigation/ui/tripprogress/TripProgressProcessor.kt

```diff
@@ -2,12 +2,53 @@ package com.mapbox.navigation.ui.tripprogress
 
 import com.mapbox.navigation.base.trip.model.RouteProgress
 import com.mapbox.navigation.base.trip.model.RouteProgressState
+import com.mapbox.navigation.utils.internal.ifNonNull
 import java.util.Calendar
 
 internal class TripProgressProcessor {
 
     fun process(action: TripProgressAction): TripProgressResult {
-        return calculateTripProgress(action as TripProgressAction.CalculateTripProgress)
+        return when (action) {
+            is TripProgressAction.CalculateTripDetails -> {
+                calculateTripDetails(action)
+            }
+            is TripProgressAction.CalculateTripProgress -> {
+                calculateTripProgress(action)
+            }
+        }
+    }
+
+    private fun calculateTripDetails(
+        action: TripProgressAction.CalculateTripDetails
+    ): TripProgressResult.TripDetails {
+        val tripDetailsList = mutableListOf<TripProgressResult.RouteLegTripDetail>()
+        ifNonNull(action.route.directionsRoute.legs()) { legs ->
```

Comment:
return an error if this check fails

Reply:
```
class TripDetailsValue internal constructor(
    val routeLegTripDetail: List<RouteLegTripDetails>,
    val totalTime: Double,
    val totalDistance: Double,
    val totalEstimatedTimeToArrival: Long,
    @ColorInt val trafficCongestionColor: Int,
    val formatter: TripProgressUpdateFormatter
)
```
Why throw an error? If legs is null we can still return `totalTime`, `totalDistance` and `totalEstimatedTimeToArrival`. In case the legs are null we can return `routeLegTripDetail` as an empty list

@LukasPaczos